src/WebJobs.Script/DependencyInjection/ExternalConfigurationStartupValidator.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.DependencyInjection
+{
+    internal class ExternalConfigurationStartupValidator : IHostedService
+    {
+        private readonly IConfigurationRoot _originalConfig;
+        private readonly IConfiguration _config;
+        private readonly IFunctionMetadataManager _metadataManager;
+        private readonly ILogger<ExternalConfigurationStartupValidator> _logger;
+
+        private static readonly IDictionary<string, KnownTrigger> _knownTriggers = new Dictionary<string, KnownTrigger>
+        {
+            { "blobTrigger", new KnownTrigger("connection", "AzureWebJobsStorage") },
+            { "queueTrigger", new KnownTrigger("connection", "AzureWebJobsStorage") },
+            { "cosmosDBTrigger", new KnownTrigger("connectionStringSetting", "ConnectionStrings:CosmosDB") }
+            // TODO: the rest
+        };
```

Comment:
May not be a valid question but just throwing it out there. Can there be settings that matter to the scale controller only for specific SKUs and some settings that may not be? Along those lines, different defaults for different SKUs?